.pre-commit-config.yaml

```
@@ -218,39 +218,34 @@ repos:
     - tests/
     pass_filenames: false
 
-- repo: local
+- repo: https://github.com/PyCQA/pylint.git
+  rev: v2.12.2
   hooks:
   - id: pylint
-    language: system
-    name: PyLint
-    files: \.py$
-    entry: python -m pylint
-    args:
-    - --ignore=tm_tokenize
-    - |-
-      --disable=
-      consider-using-with,
-      implicit-str-concat,
-      import-error,
-      invalid-name,
-      missing-class-docstring,
-      missing-function-docstring,
-      missing-module-docstring,
-      no-else-return,
-      no-member,
-      no-self-use,
-      protected-access,
-      redefined-builtin,
-      too-few-public-methods,
-      too-many-arguments,
-      too-many-branches,
-      too-many-instance-attributes,
-      too-many-locals,
-      too-many-statements,
-      unused-import,
-      useless-else-on-loop,
-      wrong-import-order,
-    stages:
-    - manual
+    additional_dependencies:
+    # NOTE: The entries below must be kept in sync with the contents of `setup.cfg` at all times,
+    # NOTE: including `install_requires` and any extras that may be imported in runtime to accommodate
+    # NOTE: for `pylint`'s ability to import the modules being checked. They must also include any of the
+    # NOTE: test and platform-dependent requirements that may be imported under different circumstances.
+    - ansible-core  # dependency imported in utils/catalog_collections.py
+    # setup.cfg:
+    - ansible-runner ~= 2.0
+    - dataclasses; python_version < "3.7"
+    - importlib-resources; python_version < "3.9.0"
+    - jinja2
+    - onigurumacffi
+    - pyyaml
+    # tests:
+    - libtmux
+    - pytest
+    # docs:
+    - sphinx
+    - setuptools-scm
```

Re-add the plugin lost by this migration:
```suggestion
    - setuptools-scm
    # plugins:
    - pylint-pytest < 1.1.0
```

---

Could this wait until the repo needs it? I think pylint is running clean without it...

---

also- I don't think, but could be wrong, the current "enforcing" pylint doesn't have this

---

We don't have this problem only because the tests are made complicated with classes. Once simplified, it's inevitable.